### ISSUE TITLE
Feature suggestion - simplification of REST API for CPTs

### ISSUE THREAD

**Issue by Xyala** [CrocoBlock/suggestions#7856]
CPTs today require potentially 2x toggles to be turned ON 
- the Advanced section REST API toggle
- the meta fields section REST API toggle

Maybe I'm not aware of the reason that would be, but I would suggest out of simplicity and to reduce support for this to have it only once, or auto-on on meta fields with the opposite behavior, a toggle to turn meta fields over API OFF.

Another related suggestion would be to have role permission on the REST endpoints.
So it's uniform with other parts of jetengine (like query) which allow to restrict endpoint access to selected user roles.
Right now we have to use plugins like AAM to limit access to CPT /CCT endpoints when it could be built-into jetengine.

**Comment by ihslimn** [comment]
> auto-on on meta fields with the opposite behavior, a toggle to turn meta fields over API OFF

Potentially, some meta fields may contain information that should not be open. Thus, it would be risky to make meta fields included into REST API by default. 

> to have role permission on the REST endpoints

This will be considered, thank you for your suggestion!

**Comment by ihslimn** [comment]
> to have role permission on the REST endpoints

For now, implementation of this feature is postponed. We will return to it when time allows. You may continue using the above-mentioned plugin to set limits for endpoints.
